test/hoot/spec/datasetPanel/tableBehavior.js

```diff
@@ -36,49 +36,57 @@ module.exports = () => {
-                folderParams  = {
-                    parentId: 0,
-                    folderName: 'UnitTestFolder1'
-                };
+                    d3.select( '.hoot-confirm .confirm-actions button.primary' ).dispatch( 'click' );
 
-            await Promise.all( _.map( layerParams, params => Hoot.api.uploadDataset( params ) ) ); // generate  test layer
-            await Hoot.api.addFolder( folderParams ); // generate test folder
-            await Hoot.folders.refreshAll();
-            await Hoot.events.emit( 'render-dataset-table' );
+                    setTimeout( () => { // wait for delete process to begin
+                        Hoot.ui.managePanel.datasets.processRequest;
+                    }, 300 );
+                }
+            });
         } );
 
+        describe( 'file ingest', () => {
+            it( 'calls uploadDataset for multiple files ingest', async function() {
+                table         = d3.select( '#dataset-table' );
+                datasetsPanel = Hoot.ui.managePanel.datasets;
 
-        after( async function() {
+                // this.timeout(30000);
 
-                ["UnitTestFolder", "UnitTestFolder1"].forEach( fName => {
+                let generateCount = 4,
+                    layerParams   = await generateOsmLayerParams( [ ...Array( generateCount ).keys() ] );
 
-                    var f = table.select( 'g[data-name="' + fName + '"]' );
-                    if (f.size()) {
-                        f.node().dispatchEvent( contextClick );
-                        d3.select( '.context-menu li:nth-child(1)' ).dispatch( 'click' );
-                        d3.select( 'body' ).dispatch( 'click' );
+                const ingest1 = await Hoot.api.uploadDataset( layerParams[ 0 ] );
+                const ingest2 = await Hoot.api.uploadDataset( layerParams[ 1 ] );
+                const ingest3 = await Hoot.api.uploadDataset( layerParams[ 2 ] );
+                const ingest4 = await Hoot.api.uploadDataset( layerParams[ 3 ] );
 
-                        confirmOverlay = d3.select( '.hoot-confirm' );
-                        d3.select( '.hoot-confirm .confirm-actions button.primary' ).dispatch( 'click' );
+                return Promise.all( [ ingest1, ingest2, ingest3, ingest4 ] );
+            } );
+        } );
 
-                        setTimeout( () => { // wait for delete process to begin
-                            Hoot.ui.managePanel.datasets.processRequest
-                        }, 300 );
-                    }
+        describe( 'folder add', () => {
+            it( 'calls addFolder', async () => {
+                const folderParams  = {
+                    parentId: 0,
+                    folderName: 'UnitTestFolder1'
+                };
 
-                });
+                await Hoot.api.addFolder( folderParams ); // generate test folder
+                const refresh = Hoot.folders.refreshAll();
 
+                return refresh;
+            } );
```

Comment:
So instead of a `before`, you move the dataset upload and folder add steps to their own test steps?

Reply:
i ran 2 tests at the end where 1 had it in the before and the latest didnt. they both seem to have passed. you think its better to keep it in before?

Reply:
i just want it to test the right behavior and pass consistently :grin:

Reply:
i think the only difference is if the before fails the stuff following it immediately fails too